presentation/src/main/java/com/pomonyang/presentation/base/BaseViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.pomonyang.presentation.base
+
+import androidx.annotation.CallSuper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+
+abstract class BaseViewModel<STATE : ViewState, EVENT : ViewEvent, EFFECT : ViewSideEffect> : ViewModel() {
+    // UI의 초기 상태를 설정
+    abstract fun setInitialState(): STATE
+
+    abstract suspend fun handleEvent(event: EVENT)
+
+    // 초기 상태를 지연 초기화
+    private val initialState: STATE by lazy { setInitialState() }
+
+    private val _state = MutableStateFlow<STATE>(initialState)
+    val state = _state.asStateFlow()
+
+    @Suppress("ktlint:standard:backing-property-naming")
+    private val _events: Channel<EVENT> = Channel<EVENT>(Channel.BUFFERED)
+
+    @Suppress("ktlint:standard:backing-property-naming")
+    private val _effects: Channel<EFFECT> = Channel<EFFECT>(Channel.BUFFERED)
+    val effect: Flow<EFFECT> = _effects.receiveAsFlow()
+
+    init {
+        subscribeToEvents()
+        onInit()
+    }
+
+    private fun subscribeToEvents() {
+        viewModelScope.launch {
+            for (event in _events) {
+                handleEvent(event)
+            }
+        }
```

Comment:
만약 `subscribeToEvents`를 유지하는 방향으로 간다면 `consumeAsFlow` 가 좀 더 나을거 같아~!

```suggestion
        viewModelScope.launch {
            _events.consumeAsFlow().collect { event ->
                handleEvent(event)
            }
        }
```